src/pytest_markdown_docs/plugin.py

```diff
@@ -159,29 +162,55 @@ def extract_code_blocks(
     tokens = mi.parse(markdown_string)
 
     prev = ""
-    for block in tokens:
+    for i, block in enumerate(tokens):
         if block.type != "fence" or not block.map:
             continue
 
         startline = block.map[0] + 1  # skip the info line when counting
         code_info = block.info.split()
 
         lang = code_info[0] if code_info else None
+        code_options = set(code_info) - {lang}
+
+        # MDX comments are put inside of a paragraph block, so we check the
+        # block two back to see if it's a comment as we the comment needs to
+        # be directly above the code block.
+        if i >= 2 and is_mdx_comment(tokens[i - 2]):
```

Comment:
Would it be messy to make this apply only for `.mdx` contexts and not regular `.md` markdown? In regular markdown the syntax isn't considered an invisible comment afaik, so it shouldn't be treated as metadata.
It would probably require us to propagate that metadata through `MarkdownTextFile` etc.

Reply:
I think I can probably make that happen.